### PR TITLE
Use DS in profile manager

### PR DESCRIPTION
**Pull request by HannesWell**
Follow-up of PR https://github.com/eclipse-m2e/m2e-core/pull/662 that is rebased to the current master and adapted to the changes that happened in the meantime.

### CODE REVIEW

File: org.eclipse.m2e.profiles.core.tests/src/org/eclipse/m2e/profiles/core/internal/management/MavenProfileManagerTest.java [hunk 1]
```diff
@@ -16,30 +16,43 @@
...
 
 
 public class MavenProfileManagerTest extends AbstractMavenProjectTestCase {
 
-  private final IProfileManager profileManager = MavenProfilesCoreActivator.getDefault().getProfileManager();
+	private static ServiceTracker<IProfileManager, IProfileManager> profileManagerTracker;
+
+	@BeforeClass
+	public static void setUpProfileManagerTracker() throws BundleException {
+		BundleContext context = FrameworkUtil.getBundle(MavenProfileManagerTest.class).getBundleContext();
```
Comment (HannesWell) [inline org.eclipse.m2e.profiles.core.tests/src/org/eclipse/m2e/profiles/core/internal/management/MavenProfileManagerTest.java:47]: This plug-ins manifest specifies `Bundle-ActivationPolicy: lazy` so the `BundleContext` should never be null here.
Furthermore I think this code will never run outside of OSGi so I also expect that `getBundle()` never returns null unless the entire framework is in a weird state.

**Review by HannesWell — COMMENTED** [review]
The build succeeded. 
@laeubi if you are fine with this change as well, I'm going to merge it.

File: org.eclipse.m2e.profiles.core.tests/src/org/eclipse/m2e/profiles/core/internal/management/MavenProfileManagerTest.java (hunk 1 above)
Comment (laeubi) [reply org.eclipse.m2e.profiles.core.tests/src/org/eclipse/m2e/profiles/core/internal/management/MavenProfileManagerTest.java:48]: For testing purpose this should be fine.

File: org.eclipse.m2e.profiles.ui/src/org/eclipse/m2e/profiles/ui/internal/actions/ProfileSelectionHandler.java [hunk 2]
```diff
@@ -72,9 +78,19 @@ public class ProfileSelectionHandler extends AbstractHandler {
 
   private static final Logger log = LoggerFactory.getLogger(ProfileSelectionHandler.class);
 
+  @Inject
+  private IProfileManager profileManager;
+
+  public ProfileSelectionHandler() {
+    BundleContext context = FrameworkUtil.getBundle(ProfileSelectionHandler.class).getBundleContext();
+    IEclipseContext serviceContext = EclipseContextFactory.getServiceContext(context);
```
Comment (HannesWell) [inline org.eclipse.m2e.profiles.ui/src/org/eclipse/m2e/profiles/ui/internal/actions/ProfileSelectionHandler.java:86]: This plug-ins manifest specifies `Bundle-ActivationPolicy: lazy` so the `BundleContext` should never be null here.
Furthermore I think this code will never run outside of OSGi so I also expect that `getBundle()` never returns null unless the entire framework is in a weird state.

File: org.eclipse.m2e.profiles.ui/src/org/eclipse/m2e/profiles/ui/internal/actions/ProfileSelectionHandler.java [hunk 3]
```diff
@@ -72,9 +78,19 @@ public class ProfileSelectionHandler extends AbstractHandler {
 
   private static final Logger log = LoggerFactory.getLogger(ProfileSelectionHandler.class);
 
+  @Inject
+  private IProfileManager profileManager;
+
+  public ProfileSelectionHandler() {
+    BundleContext context = FrameworkUtil.getBundle(ProfileSelectionHandler.class).getBundleContext();
```
Comment (HannesWell) [inline org.eclipse.m2e.profiles.ui/src/org/eclipse/m2e/profiles/ui/internal/actions/ProfileSelectionHandler.java:85]: This plug-ins manifest specifies `Bundle-ActivationPolicy: lazy` so the `BundleContext` should never be null here.
Furthermore I think this code will never run outside of OSGi so I also expect that `getBundle()` never returns null unless the entire framework is in a weird state.

File: org.eclipse.m2e.profiles.core.tests/src/org/eclipse/m2e/profiles/core/internal/management/MavenProfileManagerTest.java (hunk 1 above)
Comment (HannesWell) [reply org.eclipse.m2e.profiles.core.tests/src/org/eclipse/m2e/profiles/core/internal/management/MavenProfileManagerTest.java:48]: Ups, I actually intended to add this comment in the class below.

File: org.eclipse.m2e.profiles.ui/src/org/eclipse/m2e/profiles/ui/internal/actions/ProfileSelectionHandler.java (hunk 3 above)
Comment (laeubi) [reply org.eclipse.m2e.profiles.ui/src/org/eclipse/m2e/profiles/ui/internal/actions/ProfileSelectionHandler.java:85]: This still do not handle dynamics, we should not use the bare OSGi API when there is no need for it and we can use a DI technology instead.

Comment (HannesWell) [reply org.eclipse.m2e.profiles.ui/src/org/eclipse/m2e/profiles/ui/internal/actions/ProfileSelectionHandler.java:85]: What's then the approach you suggest?

Comment (laeubi) [reply org.eclipse.m2e.profiles.ui/src/org/eclipse/m2e/profiles/ui/internal/actions/ProfileSelectionHandler.java:85]: Sorry for confusion ... it seems it get messed up with the recent push, as long as Eclipse offers not the `EclipseContextFactory.getServiceContext(<class>);` method think we can go with this...

Comment (HannesWell) [reply org.eclipse.m2e.profiles.ui/src/org/eclipse/m2e/profiles/ui/internal/actions/ProfileSelectionHandler.java:85]: > Sorry for confusion ... it seems it get messed up with the recent push, as long as Eclipse offers not the `EclipseContextFactory.getServiceContext(<class>);` method think we can go with this...

No problem. Makes sense to use that ones available.

**Pull request merged**